wled00/e131.cpp

```diff
@@ -174,10 +174,20 @@ void handleE131Packet(e131_packet_t* p, IPAddress clientIP, byte protocol){
 
     case DMX_MODE_PRESET:       // 2 channel: [Dimmer,Preset]
       if (uni != e131Universe || availDMXLen < 2) return;
-      applyPreset(e131_data[dataOffset+1], CALL_MODE_NOTIFICATION);
-      if (bri != e131_data[dataOffset]) {
+
+      // only apply preset if value changed 
+      if (currentPreset != e131_data[dataOffset+1] && e131_data[dataOffset+1] != 0 && 
+          // only apply preset if not in playlist, or playlist changed
+          (currentPlaylist < 0 || currentPlaylist != e131_data[dataOffset+1])) { 
+        presetCycCurr = e131_data[dataOffset+1];
+        unloadPlaylist(); // applying a preset unloads the playlist
+        applyPreset(e131_data[dataOffset+1], CALL_MODE_NOTIFICATION);
+      }
+      // only change brightness if value changed
+      if (bri != e131_data[dataOffset]) {                                        
         bri = e131_data[dataOffset];
-        strip.setBrightness(bri, true);
+        strip.setBrightness(scaledBri(bri), false);
+        stateUpdated(CALL_MODE_WS_SEND);
```

Comment:
Is this strictly necessary?
As `applyPreset()` is asynchronous in 0.14 and beyond it will trigger "CALL_MODE_NOTIFICATION" when the preset is actually applied (which is *after* `strip.setBrighntess()` call).

Reply:
Without the call, the brightness slider in the UI is _never_ updated. 
With this line tho, the UI slider updates async (without lagging the ESP to death), and then the slider represents the actual brightness state as per DMX values. 
The only thing I'm not sure of, is the `strip.setBrightness` direct _true/false_ bool - I set it to false to adhere to the refresh cycles instead of writing directly to the busses - so there might be a few ms delay but that seemed ok in my tests, and it felt more stable.